lms/djangoapps/commerce/views.py

```diff
@@ -103,28 +103,27 @@ def post(self, request, *args, **kwargs):  # pylint: disable=unused-argument
 
         # Make the API call
         try:
-            order_number, order_status, _body = api.create_order(user, honor_mode.sku)
-            if order_status == OrderStatus.COMPLETE:
-                msg = Messages.ORDER_COMPLETED.format(order_number=order_number)
+            response_data = api.create_basket(
+                user,
+                honor_mode.sku,
+                payment_processor="cybersource",
+            )
+            if response_data['payment_data'] is not None:
+                # it is time to start the payment flow.
+                return JsonResponse(response_data['payment_data'])
+            elif response_data['order']:
+                # the order was completed immediately because there was no charge.
+                msg = Messages.ORDER_COMPLETED.format(order_number=response_data['order']['number'])
                 log.debug(msg)
                 return DetailResponse(msg)
             else:
-                # TODO Before this functionality is fully rolled-out, this branch should be updated to NOT enroll the
-                # user. Enrollments must be initiated by the E-Commerce API only.
-                self._enroll(course_key, user)
-                msg = u'Order %(order_number)s was received with %(status)s status. Expected %(complete_status)s. ' \
-                      u'User %(username)s was enrolled in %(course_id)s by LMS.'
-                msg_kwargs = {
-                    'order_number': order_number,
-                    'status': order_status,
-                    'complete_status': OrderStatus.COMPLETE,
-                    'username': user.username,
-                    'course_id': course_id,
-                }
-                log.error(msg, msg_kwargs)
-
-                msg = Messages.ORDER_INCOMPLETE_ENROLLED.format(order_number=order_number)
-                return DetailResponse(msg, status=HTTP_202_ACCEPTED)
+                # TODO how does the user recover from this?
```

Comment:
On the logistration page, which auto-enrolls the user in a free track, we don't display any errors.  The assumption is that it's important to successfully authenticate the user and get them to a track selection page where they might continue to payment, and we don't want to block that process.

On the payment page, I believe we show an error message (using ErrorModel and ErrorView defined using BackBone views and models).  There probably isn't a whole lot the user can do in this case except to contact support.